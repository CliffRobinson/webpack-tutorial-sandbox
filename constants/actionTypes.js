module.exports = {
    REQUEST_MESSAGES: "REQUEST_MESSAGES",
    RECEIVE_MESSAGES: "RECEIVE_MESSAGES",
    GET_MESSAGES: "GET_MESSAGES",
    UPDATE_CURRENT_MESSAGE: "UPDATE_CURRENT_MESSAGE",
    ADD_MESSAGE: "ADD_MESSAGE",
}