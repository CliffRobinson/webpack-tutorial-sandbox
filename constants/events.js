module.exports = {
    //Chat message events
    REQUEST_MESSAGES_BY_ROOM: "REQUEST_MESSAGES_BY_ROOM",
    RECEIVE_MESSAGES: "RECEIVE_MESSAGES",
    GET_MESSAGES: "GET_MESSAGES",
    UPDATE_CURRENT_MESSAGE: "UPDATE_CURRENT_MESSAGE",
    ADD_MESSAGE: "ADD_MESSAGE",
    //Game list events
    REQUEST_GAMES_BY_STATUS: "REQUEST_GAMES_BY_STATUS",
    RECEIVE_GAMES_BY_STATUS: "RECEIVE_GAMES_BY_STATUS",
    CREATE_GAME: "CREATE_GAME",
    UPDATE_GAME: "UPDATE_GAME",
    DELETE_GAME: "DELETE_GAME"
}