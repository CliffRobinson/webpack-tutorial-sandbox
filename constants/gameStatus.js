module.exports = {
    PENDING:"pending",
    PLAYING:"playing",
    CLOSED:"closed",
    CREATING: 'creating'
}