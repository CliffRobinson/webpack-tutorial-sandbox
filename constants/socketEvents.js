module.exports = {
    REQUEST_MESSAGES_BY_ROOM: 'requestMessagesByRoom',
    ADD_CHAT_MESSAGE: 'addChatMessage'
}