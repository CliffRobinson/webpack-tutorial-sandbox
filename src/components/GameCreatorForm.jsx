import React from 'react';

import { PENDING, CREATING } from '../../constants/gameStatus';

const styleObj = {
    backgroundColor: 'white',
    border: "2px solid black"
}

export function GameCreatorForm(props) {
    const { createGame, receiveGamesByStatus } = props;
    const {creating} = props.games

    function clicker(e) {
        e.preventDefault()
        const game = {
            name: creating.name,
            password: creating.password,
            status: PENDING
        }
        createGame(game)
    }

    function inputChange(e) {
        e.preventDefault()

        const updatedCreating = {
            ...creating,
            [e.target.name]:e.target.value
        }

        console.log(updatedCreating)

        receiveGamesByStatus({
            status: CREATING,
            games: updatedCreating
        })
    }

    return(<form style={styleObj}>
        <h4> Wanna make a gaem?</h4>
        <label htmlFor='name'>Name:</label>
        <input type='text' name='name' onChange={inputChange} value={creating.name}></input>
        <label htmlFor='password'>Password:</label>
        <input type='text' name='password' onChange={inputChange} placeholder="leave blank if you don't want a password"></input>
        <button type='submit' onClick={clicker}>Let's PLAAAY</button>
    </ form>)
}