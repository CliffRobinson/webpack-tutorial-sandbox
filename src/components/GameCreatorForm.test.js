import React from 'react';
import {shallow, mount} from 'enzyme';

import { CREATE_GAME } from '../../constants/events';

import { GameCreatorForm } from './GameCreatorForm'
import { CREATING } from '../../constants/gameStatus';

const mockCreateGame = jest.fn();
const mockReceiveGamesByStatus = jest.fn()

const testProps = {
    games: {
        creating: {
            name: "",
            password: ""
        }
    },
    createGame: mockCreateGame,
    receiveGamesByStatus: mockReceiveGamesByStatus
}

test('snapshot', () => {
    const wrapper = shallow(<GameCreatorForm {... testProps}/>)

    expect(wrapper).toMatchSnapshot()
})

test('clicker', () => {
    const wrapper = mount(<GameCreatorForm {... testProps}/>)

    const testButton = wrapper.find('button').first()

    testButton.simulate('click')

    expect(mockCreateGame).toBeCalled()

    mockCreateGame.mockClear()
})

test('name onChange', () => {
    const wrapper = mount(<GameCreatorForm {... testProps}/>)

    const mockPreventDefault = jest.fn()

    const newName = 'a'

    const event = {
        preventDefault: mockPreventDefault,
        target: {
            name: 'name',
            value: newName,
        }
    }

    const nameInput = wrapper.find('input').first();

    nameInput.simulate('change', event)

    expect(mockPreventDefault).toBeCalled()
    expect(mockReceiveGamesByStatus).toBeCalledWith({
        status: CREATING,
        games: {
            name: newName,
            password: ""
        },
    })

    mockPreventDefault.mockClear()
    mockReceiveGamesByStatus.mockClear()
});

test('password onChange', () => {
    const wrapper = mount(<GameCreatorForm {... testProps}/>)

    const mockPreventDefault = jest.fn()

    const newPassword = 'a'

    const event = {
        preventDefault: mockPreventDefault,
        target: {
            name: 'password',
            value: newPassword
        }
    }

    const nameInput = wrapper.find('input').at(1);

    nameInput.simulate('change', event)

    expect(mockPreventDefault).toBeCalled()
    expect(mockReceiveGamesByStatus).toBeCalledWith({
        status: CREATING,
        games: {
            name: "",
            password: newPassword
        },
    })

    mockPreventDefault.mockClear()
    mockReceiveGamesByStatus.mockClear()
});