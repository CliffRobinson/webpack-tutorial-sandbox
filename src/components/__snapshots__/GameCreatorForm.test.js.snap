// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<form
  style={
    Object {
      "backgroundColor": "white",
      "border": "2px solid black",
    }
  }
>
  <h4>
     Wanna make a gaem?
  </h4>
  <label
    htmlFor="name"
  >
    Name:
  </label>
  <input
    name="name"
    onChange={[Function]}
    type="text"
    value=""
  />
  <label
    htmlFor="password"
  >
    Password:
  </label>
  <input
    name="password"
    onChange={[Function]}
    placeholder="leave blank if you don't want a password"
    type="text"
  />
  <button
    onClick={[Function]}
    type="submit"
  >
    Let's PLAAAY
  </button>
</form>
`;
